### PR TITLE
for V4.x.x remove future

### PR DESCRIPTION
**Pull request by hanasakari**
See https://github.com/vert-x3/vertx-starter/issues/110

### CODE REVIEW

**Comment by hanasakari** [comment]
for V4.x.x remove future (https://github.com/eclipse-vertx/vert.x/issues/2869)

File: src/main/resources/templates/src/main/java/MainVerticle.java.ftl [hunk 1]
```diff
@@ -1,12 +1,20 @@
...
+<#if vertxVersion?index_of("3") != 0>
+import io.vertx.core.Promise;
+<#else>
 import io.vertx.core.Future;
+</#if>
 
 public class MainVerticle extends AbstractVerticle {
 
   @Override
+  <#if vertxVersion?index_of("3") != 0>
```
Comment (tsegismont) [inline src/main/resources/templates/src/main/java/MainVerticle.java.ftl]: This test would be more robust with [`starts_with`](https://freemarker.apache.org/docs/ref_builtins_string.html#ref_builtin_starts_with)

File: src/main/resources/templates/src/main/java/MainVerticle.java.ftl [hunk 2]
```diff
@@ -1,12 +1,20 @@
 package ${packageName};
 
 import io.vertx.core.AbstractVerticle;
+<#if vertxVersion?index_of("3") != 0>
```
Comment (tsegismont) [inline src/main/resources/templates/src/main/java/MainVerticle.java.ftl]: This test would be more robust with [`startsWith`](https://freemarker.apache.org/docs/ref_builtins_string.html#ref_builtin_starts_with)

File: src/main/resources/templates/src/main/java/MainVerticle.java.ftl [hunk 3]
```diff
@@ -1,12 +1,20 @@
...
+import io.vertx.core.Promise;
+<#else>
 import io.vertx.core.Future;
+</#if>
 
 public class MainVerticle extends AbstractVerticle {
 
   @Override
+  <#if vertxVersion?index_of("3") != 0>
+  public void start(Promise<Void> startFuture) throws Exception {
```
Comment (tsegismont) [inline src/main/resources/templates/src/main/java/MainVerticle.java.ftl]: Can you rename the parameter to `startPromise`?

Comment (hanasakari) [reply src/main/resources/templates/src/main/java/MainVerticle.java.ftl]: ok

**Pull request merged**